### PR TITLE
Add `HistoryFactory`

### LINKED ISSUE #70: Add `History` factory
## 📚 Context

Currently, if one wants to deploy a `History` contract, the easiest way is through smart contract frameworks like Hardhat or Foundry. Still, this is not trivial for most web3 users, thus limiting the number of validators that can deploy their own Consensus contracts and associated History contracts. Furthermore, the DApp developer has to trust that the validator didn't deploy a malicious implementation of a consensus contract.

## ✔️ Solution

We assume that validators are able to sign and submit transactions to the blockchain. Therefore, a safe and easy way for them to deploy a History contract is by submitting a message call to a factory contract that deploys a new History contract.

## 📈 Subtasks

- [ ] Create a PR that adds a `History` factory and associated deployment script
- [ ] Merge the PR

### PR DESCRIPTION
**Pull request by guidanoli**
Adds interface, implementation and tests.

Closes cartesi/rollups-contracts#27.

### CODE REVIEW

File: onchain/rollups/contracts/history/HistoryFactory.sol [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+pragma solidity ^0.8.8;
+
+import {Create2} from "@openzeppelin/contracts/utils/Create2.sol";
+
+import {IHistoryFactory} from "./IHistoryFactory.sol";
+import {History} from "./History.sol";
+import {IInputBox} from "../inputs/IInputBox.sol";
```
[the remark this answers is not in the document]
Comment (guidanoli) [reply onchain/rollups/contracts/history/HistoryFactory.sol:19]: Oops! Thanks!

**Review by ZzzzHui — APPROVED** [review]

**Review by pedroargento — APPROVED** [review]

**Comment by guidanoli** [comment]
Rebased.

**Comment by guidanoli** [comment]
Rebased and organized commits.

**Comment by guidanoli** [comment]
Re-committed with the correct e-mail.

**Comment by guidanoli** [comment]
This PR has been migrated to https://github.com/cartesi/rollups-contracts/pull/28.

**Pull request closed without merging**